### PR TITLE
Fix/reload postings view bug

### LINKED ISSUE #14: User should not be able to add a posting if already added to upcoming jobs
As a user, you should not be able to add a posting to your list if the job has already been added to your list. Additionally, the postings view should not show an event as upcoming and open if the open positions is 0 or less.

### LINKED ISSUE #21: Posting view page reload bug
If a user refreshes the posting view page there is an error that occurs.

### LINKED ISSUE #22: Submit button should not add from an event that has already been added
As a user, I should not be able to add twice from a specific event... The submit button should be disabled if a user has an upcoming job from the given event

### PR DESCRIPTION
**Pull request by coopterrones**
#### What's this PR do?
-Disables submit button if posting has already been added to the users profile

#### What does it fix?
-posting view bug on reload

#### Where should the reviewer start?
-line 5 `postingView.js`
-line 26 `App.js`

#### How should this be manually tested?
-Test that bugs are no longer apparent. When in postingView.js reload the page and it should load with no wrrors.
-Test that you cannot submit once a user already has a posting from an event in upcoming jobs

### CODE REVIEW

File: src/Components/App/index.js [hunk 1]
```diff
@@ -23,9 +23,13 @@ const App = () => {
 
   const searchPostings = (keyWord) => {
     const lowerCaseKeyword = keyWord.toLowerCase();
-    const filteredPostings = postings.filter(posting => posting.name.toLowerCase().includes(lowerCaseKeyword) || posting.organization.toLowerCase().includes(keyWord));
+    const filteredPostings = postings.filter(
+      (posting) =>
```
Comment (yiranli624) [inline src/Components/App/index.js:27]: Nice use of condition

File: src/Components/PostingView/index.js [hunk 2]
```diff
@@ -72,14 +82,24 @@ const PostingView = ({ eventId, getUserInfo }) => {
...
+                >
                   <h3>{job.name}</h3>
                   <p>Open Spots: {job.numberOfSpots}</p>
                 </section>
               ))}
             </div>
             <div className="submit-button-wrapper">
-              <button onClick={substractOpenPosition} className="submit-button">SUBMIT</button>
+              <button
+                onClick={substractOpenPosition}
```
Comment (okalimbet) [inline src/Components/PostingView/index.js:104]: good feature! It's nice to show users when they are not allowed to choose more position by disabling the button. 
We might show specific effect/message that shows the confirmation of the position and that user already signed for this event.

File: src/Components/App/index.js [hunk 3]
```diff
@@ -34,8 +38,17 @@ const App = () => {
...
+        exact
+        path="/"
+        render={() => (
+          <Postings
+            postings={searchedPostings.length ? searchedPostings : postings}
+            searchByKeyWord={searchPostings}
+          />
+        )}
+      />
+      <Route path="/postings/:id" component={PostingView} />
```
Comment (yiranli624) [inline src/Components/App/index.js:51]: Great work with Route render and component!

**Review by okalimbet — APPROVED** [review]

**Review by yiranli624 — APPROVED** [review]

File: src/Components/PostingView/index.js (hunk 2 above)
Comment (yiranli624) [reply src/Components/PostingView/index.js:104]: good idea! and add affect to reflect selection

**Pull request merged**